doc/feature/restAPI.md

```diff
@@ -0,0 +1,94 @@
+Publishing the REST API exposed by the HTTP Server
+============
+
+The http server exposes C++ objects via a REST API, in a JSON serialized format.
+The purpose of this document is to describe how this API request is implemented,
+and invoked.
+
+## Requirements
+
+* Define how to invoke the application REST API
+* Implement the JSON representation of the REST API
+
+## API
+
+Invoking a GET request on the / url returns the list of top-level namespaces, 
+e.g. 'zerobuf'. Similarily, a GET /zerobuf/render/ would return the objects in
+the zerobuf::render namespace. Finally, a GET /namespace/object/schema would 
+return the schema for the object itself.
+
+   Request: GET http://hostname:port/
+   Response: [ "namespace1", "namespace2", "..." ]
+
+   Request: GET http://hostname:port/namespace1/
+   Response: [ "object1", "object2", "..." ]
+
+   Request: GET http://hostname:port/namespace1/object1/schema
+   Response: { "object1" : { "attribute1": "string", "attribute2": "boolean" } }
+
+## Examples
+- Create an HTTP server and register serializables
+
+    void setupHTTPServer( const int argc, char** argv )
+    {
+        // Create HTTP server
+        _httpServer = zeq::http::Server::parse( argc, argv );
+        if( !_httpServer )
+            return;
+
+        // Register C++ objects
+        subscribers.push_back( _httpServer );
+        _httpServer.subscribe( camera, LUT, vrParams );
+        _httpServer.register( camera, LUT, vrParams, frame );
+    }
+
+- Query API
+   Request: GET http://hostname:port/zerobuf/render/camera/schema
+   Response: { "camera" : { "origin": "string", "lookat": "up" } }
+
+## Implementation
+
+* Add a getSchema public method to Servus::serializable
+* Make sure getSchema is added to code generator
+* Add the '/schema' url to every registered object, and make it return the
+  schema of the object in a JSON form.
+
+## Issues
+
+### API
+
+Several options are beging discussed regarding the implementation of the API:
+
+#### Option 1
+GET / returns the list of top-level namespaces, e.g., 'zerobuf', GET 
+/zerobuf/render/ returns the objects in zerobuf::render. GET 
+/namespace/object/schema returns the schema for the object.
+- Example:
+   Request: GET http://hostname:port/
+   Response: [ "namespace1", "namespace2", "..." ]
+
+   Request: GET http://hostname:port/namespace1/
+   Response: [ "object1", "object2", "..." ]
+
+   Request: GET http://hostname:port/namespace1/object1/schema
+   Response: { "object1" : { "attribute1": "string", "attribute2": "boolean" } }
+- Pros: Clean API, and a one-to-one match between a URI and the registered 
+  object.
+- Cons: Client needs to traverse the tree or URIs to get the list of registered 
```

Comment:
If the specification defines that there is always a namespace after host:port, there is no ambiguity anymore.